src/fondant/component_spec.py

```diff
@@ -102,6 +167,23 @@ class ComponentSpec:
     def __init__(self, specification: t.Dict[str, t.Any]) -> None:
         self._specification = copy.deepcopy(specification)
         self._validate_spec()
+        self._type: ComponentType = self._get_component_type()
+
+    def _get_component_type(self) -> "ComponentType":
+        """Function that returns the component type based on the component specification."""
+        consumes = self._specification.get("consumes", False)
+        produces = self._specification.get("produces", False)
+
+        if not consumes and produces:
+            _type = ComponentType.LOAD
+        elif consumes and produces:
+            _type = ComponentType.TRANSFORM
+        elif consumes and not produces:
+            _type = ComponentType.WRITE
```

Comment:
I think this is the case for filtering components as well. Eg. https://github.com/ml6team/fondant/blob/main/components/filter_comments/fondant_component.yaml

Reply:
Do we need to infer it from the spec only or can we also use the `Component` class? Because there it is already indicated if it is a Load, Transform, or Write component.

Reply:
Nvm, I don't think that's a good direction. You're using this info in the compiler and pipeline, where we're working with the spec only.